src/compute/pre_compute_app.rs

```diff
@@ -189,13 +189,10 @@ impl PreComputeAppTrait for PreComputeApp {
         .ok_or(ReplicateStatusCause::PreComputeDatasetDownloadFailed)?;
 
         info!("Checking encrypted dataset checksum [chainTaskId:{chain_task_id}]");
-        let expected_checksum = args
-            .encrypted_dataset_checksum
-            .as_ref()
-            .ok_or(ReplicateStatusCause::PreComputeDatasetDownloadFailed)?;
+        let expected_checksum: &str = &args.encrypted_dataset_checksum;
         let actual_checksum = sha256_from_bytes(&encrypted_content);
 
-        if actual_checksum != *expected_checksum {
+        if actual_checksum != args.encrypted_dataset_checksum {
```

Comment:
in this case here, as it is used just below, I think we could keep the `expected_checksum` variable here